kapitan/targets.py

```diff
@@ -100,21 +99,21 @@ def compile_targets(inventory_path, search_paths, output_path, parallel, targets
 
         # Save inventory and folders cache
         save_inv_cache(compile_path, targets)
-
+        pool.close()
```

Comment:
Can you test this doesn't impact speed with multiple `-p` options in compile, I feel like this might have unintended consequences

Reply:
Why close here?

Reply:
tested with -p 16 and -p 8, output is the same as latest kapitan.

Reply:
But why?

Reply:
Because we don't have more work to submit after that point (?) https://stackoverflow.com/questions/38271547/when-should-we-call-multiprocessing-pool-join I might be wrong, it's just what I understood from here ^ and what I saw in other examples of pool usage.